platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/CipherUtils.java

```diff
@@ -0,0 +1,205 @@
+import static org.apache.commons.codec.binary.Hex.encodeHex;
+import static org.apache.commons.io.FileUtils.writeStringToFile;
+import static org.codice.ddf.configuration.migration.MigrationZipConstants.getDefaultChecksumPathFor;
+import static org.codice.ddf.configuration.migration.MigrationZipConstants.getDefaultKeyPathFor;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Charsets;
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import javax.crypto.Cipher;
+import javax.crypto.CipherOutputStream;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.migration.MigrationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CipherUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CipherUtils.class);
+
+  private final Path zipPath;
+  private final Path keyPath;
+  private final Path checksumPath;
+  private final Cipher cipher;
+  private SecretKey secretKey;
+
+  public CipherUtils(Path zipPath) {
+    this(zipPath, getDefaultKeyPathFor(zipPath), getDefaultChecksumPathFor(zipPath));
+  }
+
+  public CipherUtils(Path zipPath, Path keyPath, Path checksumPath) {
+    Validate.notNull(zipPath, "Null zip path");
+    Validate.notNull(keyPath, "Null key path");
+    Validate.notNull(checksumPath, "Null checksum path");
+    this.zipPath = zipPath;
+    this.keyPath = keyPath;
+    this.checksumPath = checksumPath;
+    cipher = createEncryptionCipher();
+  }
+
+  /**
+   * Wraps a given {@link OutputStream} in a {@link CipherOutputStream}
+   *
+   * @param outputStream
+   * @return a {@link CipherOutputStream} for writing encrypted {@link java.util.zip.ZipEntry}
+   *     contents
+   */
+  public CipherOutputStream getCipherOutputStream(OutputStream outputStream) {
+    return new CipherOutputStream(outputStream, cipher);
+  }
+
+  /**
+   * Creates a {@link Cipher} for use during file encryption and stores the secret key in a file
+   *
+   * @throws MigrationException when an invalid key is provided
+   * @throws MigrationException when an invalid key padding is used
+   * @throws MigrationException when and invalid cipher algorithm is used
+   * @return a {@link Cipher} in encrypt mode
+   */
+  private Cipher createEncryptionCipher() {
```

Comment:
✏️ Nit: public methods all to the top, then protected/package-private, and then finally private. You've got this here, some public methods below it, then a bunch of private methods, then finally more public.

Reply:
Done